components/data_prefetch/index.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {ActionCreatorsMapObject, bindActionCreators, Dispatch} from 'redux';
+
+import {getCurrentChannelId, getUnreadChannels} from 'mattermost-redux/selectors/entities/channels';
+import {memoizeResult} from 'mattermost-redux/utils/helpers';
+import {isChannelMuted} from 'mattermost-redux/utils/channel_utils';
+import {getMyChannelMemberships} from 'mattermost-redux/selectors/entities/common';
+
+import {Channel, ChannelMembership} from 'mattermost-redux/types/channels';
+
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+import {RelationOneToOne} from 'mattermost-redux/types/utilities';
+
+import {GlobalState} from 'types/store';
+
+import {prefetchChannelPosts} from 'actions/views/channel';
+
+import DataPrefetch from './data_prefetch';
+
+type Actions = {
+    prefetchChannelPosts: (channelId: string, delay: number | undefined) => Promise<{data: {}}>;
+};
+
+const prefetchQueue = memoizeResult((channels: Channel[], memberships: RelationOneToOne<Channel, ChannelMembership>) => {
```

Comment:
Can you add a comment explaining what this does?

Are we concerned if this returns a new object that's equal to the old one? Like renaming a channel or changing some settings will cause this to return a new, identical object.

Reply:
It is not for `channel` obj or memberships but in gerenal state changes. Selector felt like overkill and the memoization is just as a precaution.